frontend/lib/resolution/Resolver.cpp

```diff
@@ -2629,7 +2632,26 @@ bool Resolver::enter(const NamedDecl* decl) {
 
   enterScope(decl);
 
-  return true;
+  // TODO: Initializers will need to check the compatibility of the type
+  // expression and initialization expression.
+  if (!scopeResolveOnly && fieldTypesOnly && decl == curStmt) {
+    auto field = decl->toVarLikeDecl();
+    if (field->typeExpression() != nullptr &&
```

Comment:
AFAIK you could also make this condition check `field->initExpression() != nullptr`, right? And then not check it below? IMO that would be a bit nicer.

Anyway it'd be good to put a comment about why this logic is here, separately from the TODO